asreview/project.py

```diff
@@ -59,60 +63,126 @@ class ProjectNotFoundError(Exception):
     pass
 
 
-def get_project_path(project_id, asreview_dir=None):
+# !=============================================
+# @TODO: I HATE THE FACT THAT I AM TALKING ABOUT
+# Users in this module, but I need a quick fix
+# for now
+# !=============================================
+
+
+def get_project_path(project_id, user=None, asreview_dir=None):
     """Get the project directory.
 
     Arguments
     ---------
     project_id: str
         The id of the current project.
+    user: User
+        User account object.
     """
 
     if asreview_dir is None:
         asreview_dir = asreview_path()
 
-    return Path(asreview_dir, project_id)
+    if isinstance(user, User) and isinstance(user.id, int):
+        # if we do authentication, then the project must be 
+        # registered in the database
+        project_from_db = Project.query.filter(
+            Project.project_id == project_id
+        ).one_or_none()
 
+        # project exists and user -is- owner OR this is a new project
+        if (project_from_db and user == project_from_db.owner) or (
+            project_from_db == None
```

Comment:
`project_from_db is None` is preferred over `project_from_db == None`. In theory a class can have custom implementation of the comparison operator and here we are really want to know if the value is None.